### PR TITLE
Flycheck

### PR DESCRIPTION
**Pull request by nosami**
Making flycheck using the new stdio interface of the server. No more curl!

### CODE REVIEW

File: test/buttercup-tests/flycheck/flycheck-test.el [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(describe "Flycheck integration"
+  (it "renames a symbol referenced only in a single file"
```
Comment (sp3ctum) [inline test/buttercup-tests/flycheck/flycheck-test.el]: Copy paste residue :)

**Comment by sp3ctum** [comment]
Awesome work :+1:

File: test/buttercup-tests/flycheck/flycheck-test.el [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    (ot--buffer-contents-and-point-at-$
+     "$public class MyClass {
+         DoesNotExist foo;
+       }")
+
+    (flycheck-mode)
+    (flycheck-list-errors)
+    ;; (ot--wait-for (lambda() (print "waiting")(get-buffer "*Flycheck errors*")) 5)
+    (ot--wait-for (lambda() (and (boundp 'flycheck-current-errors)
+                                 (> (length flycheck-current-errors) 0))) 10)
```
Comment (nosami) [inline test/buttercup-tests/flycheck/flycheck-test.el]: This line never finds any errors when ran from the integration tests script - I can't figure out why. I tried quite a few other ways of achieving the same thing but no luck. 
It's odd - the commented out look for 'Flycheck buffer' line above doesn't fail, so I can see that flycheck mode is running.

The test passes every time when ran locally using `M-x buttercup-run-at-point`

@sp3ctum Any chance that you could take a look at this for me? I'd appreciate it.

Comment (sp3ctum) [inline test/buttercup-tests/flycheck/flycheck-test.el]: Sure, I can take a look at it. Maybe flycheck itself has some trick for
running tests reliably.
4.8.2015 5.08 ap. "Jason Imison" <notifications@github.com> kirjoitti:

**Pull request merged**